lib/determine-basal/cob.js

```diff
@@ -38,15 +44,8 @@ function detectCarbAbsorption(inputs) {
         } else if (glucose_data[i].dateString) {
             bgTime = new Date(glucose_data[i].dateString);
         } else { console.error("Could not determine BG time"); }
-        if (bucketed_data[bucketed_data.length-1].display_time) {
-            lastbgTime = new Date(bucketed_data[bucketed_data.length-1].display_time.replace('T', ' '));
-        } else if (glucose_data[i-1].display_time) {
-            lastbgTime = new Date(glucose_data[i-1].display_time.replace('T', ' '));
-        } else if (glucose_data[i-1].dateString) {
-            lastbgTime = new Date(glucose_data[i-1].dateString);
-        } else { console.error("Could not determine last BG time"); }
-        if (! glucose_data[i].glucose || glucose_data[i].glucose < 39 || glucose_data[i-1].glucose < 39) {
```

Comment:
Why did you remove this instead of modifying it?

Reply:
Lines 41-47 were moved lower and lines 48 and 49 were modified to remove the check against the glucose_data[i-1].glucose value and are now lines 47 and 48.

The [i-1] check is no longer required due to how lastbgi is managed.